Разметка всех страниц

AleksanrReva opened the pull request:
Выполнил задание 4 из раздела «2.42. Разметка всех страниц»


---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss-individual/1/user/1411729/tasks/4)

:boom: https://htmlacademy-htmlcss.github.io/1411729-device-1/2/

AleksanrReva committed:
Добавляет хеадер

Только разметка основного содержимого: логотип, строка поиска, навигация первого уровня

AleksanrReva committed:
Добавляет слайдер

Разметка промо-слайдера на примере одного слайда

AleksanrReva committed:
Добавляет меню каталога

Разметка меню каталога.

AleksanrReva committed:
Добавляет Информацию об услугах

Разметка блока в котором содержится информация о Доставке, Гарантиях и Кредите

AleksanrReva committed:
Добавляет "баннер" доставки под заказ

Разметка блока для вызова модального окна "Доставка под заказ"

AleksanrReva committed:
Добавляет две колонки О нас и Контакты

Разметка блока О нас и Контакты

AleksanrReva committed:
Добавляет Подписку

Разметка блока с формой отправки адреса электронной почты клиента для подписки

AleksanrReva committed:
Добавляет Футер

Классы, где есть, могут не совпадать с содержимым. Основная разметка сделана

AleksanrReva committed:
Вносит корректировки в header

Логотип вынесен за <nav>. Так последний не выглядит оберткой-прослойкой и для верстки проще, имхо. Поправил название классов у формы и задал класс ей самой

AleksanrReva committed:
Вносит корректировки в main

В основном изменения коснулись названий классов. Добавлены символы для кнопок прокрутки фотографий в слайдере; линии для Контактов и О нас

AleksanrReva committed:
Вносит корректировки в footer

Работа с названиями классов

AleksanrReva committed:
Добавляет catalog.html

Создаёт базовую разметку страницы

AleksanrReva committed:
Добавляет хеадер страницы

Разметка элементов хеадера

AleksanrReva committed:
Добавляет Фильтр каталога

Разметка колонки с настройками фильтра сортировки товара

AleksanrReva committed:
Добавляет товары в каталог

Заменена ссылка на кнопку в Фильтре каталога. Размечена область отображения карточек товара каталога

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,348 @@
+          <li class="navigation-user-item">
+
+            <form class="product-search" action="https://echo.htmlacademy.ru/" method="get">
+              <input class="product-search-name" type="text" name="productName" value="Поиск по сайту">
```
для input нужен label.  атрибуты не нужно писать в стиле camelCase. https://up.htmlacademy.ru/htmlcss-individual/1/criteries#PROJ-02

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,348 @@
+          <fieldset class="catalog-filter-group">
+            <legend class="catalog-filter-title">Стоимость</legend>
+
+            <input class="catalog-filter-price" type="range" name="price" min="0" max="9000" step="50" value="5000">
```
для всех инпутов label

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,348 @@
+                  Розовый
+                </label>
+              </li>
+            </ul>
```
у всей группы чекбоксов одинаковый name, уникальный value. Аналогично радиокнопкам

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,348 @@
+
+            <ul class="catalog-filter-list">
+              <li class="catalog-filter-item">
+                <lable class="catalog-filter-name">
```
lable -  такого элемента нет, label. исправить нужно во всем документе

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,348 @@
+          </select>
+        </label>
+
+        <ul class="sort-order-list">
+          <li class="sort-order-item">
+            <lable class="sort-order-name">
+              <input class="sort-order-control" type="radio" name="order" value="ascending" checked>
+              &#9650;
+            </lable>
+          </li>
+
+          <li class="sort-order-item">
+            <lable class="sort-order-name">
+              <input class="sort-order-control" type="radio" name="order" value="descending">
+              &#9660;
+            </lable>
+          </li>
+        </ul>
```
это две кнопки для сортировки

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,348 @@
+          Санкт-Петербург, набережная реки Карповки, 5, литера П.
+        </address>
+
+        <hr class="footer-address-line">
```
нужно убрать из разметки, будет псевдоэлемент в стилях

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,348 @@
+        <hr class="footer-address-line">
+      </section>
+
+      <section class="footer-navigation">
```
для всех секций нужен заголовок

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+          Заодно сможете проверить работоспособность покупки.
+        </p>
+
+        <address class="info-text contacts-address">
+          <h3 class="info-title">Интернет-магазин «Девайс»</h3>
+
+          Санкт-Петербург, набережная реки Карповки, 5, литера П.
+
+          <a href="tel:+78128121212">+7 (812) 812-12-12</a>
+        </address>
+
+        <p class="info-text">
+          <h3 class="info-title">Время работы</h3>
+          пн—пт: с 10:00 до 22:00
+          сб—вс: с 10:00 до 19:00
+        </p>
```
заголовки нельзя вкладывать в address, в параграф нельзя вкладывать заголовки

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+    </header>
+
+    <main class="page-index">
+      <h1 class="visually-hidden">Каталог товаров</h1>
```
это главная страница, у нее главный заголовок не должен быть "каталог товаров"

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+          <h3 class="visually-hidden">Фотографии</h3>
+
+          <button class="product-photo-previos" type="button" value="photoPrevios">
+            <span class="visually-hidden">&#10094;</span >
```
здесь должно быть пояснение к кнопке, не символ

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+        <section class="product-photo">
+          <h3 class="visually-hidden">Фотографии</h3>
+
+          <button class="product-photo-previos" type="button" value="photoPrevios">
```
value тут не нужен

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+
+          <label class="services-name">
+
+            <input class="services-menu-item" type="radio" name="serviceName" checked>
```
написание атрибутов

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+        </div>
+      </section>
+
+      <section class="banner-delivery">
+        <h2 class="visually-hidden">Заказ товара</h2>
+
+        <a class="banner-delivery-link" href="#">ДОСТАВИМ РЕДКИЙ ТОВАР ПОД ЗАКАЗ</a>
+
+        <button class="banner-delivery-tooltip" aria-label="Краткая информация о доставке">
+          i
+        </button>
+      </section>
```
это все одна ссылка

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+      </section>
+
+      <section class="info about-us">
+        <hr class="info-indent-line">
```
hr не нужен

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+      </section>
+
+      <section class="info contacts">
+        <hr class="info-indent-line">
```
hr не нужен

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+          а также акции, спецпредложения и вот это вот всё.
+        </p>
+
+        <form class="subscribe-email-send" action="https://echo.htmlacademy.ru/" method="get">
```
method="post"

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,375 @@
+
+        <form class="subscribe-email-send" action="https://echo.htmlacademy.ru/" method="get">
+
+          <input class="subscribe-email" type="email" name="subscribeEMail" value="Ваш e-mail">
```
атрибуты наименование

AleksanrReva committed:
Добавляет label, исправляет тэги и name

Добавил, недостающие label для input, исправил "lable".
Здесь же, убрал camelCase в атрибутах

AleksanrReva committed:
Исправляет атрибуты чекбоксов

Задал одно имя для чекбоксов, а предыдущие значения перезаписал в value

AleksanrReva committed:
Заменяет radio input на button

AleksanrReva committed:
Убирает hr

Удалил все линии

AleksanrReva committed:
Обновляет слайдер

Убрал атрибут value, символы стрелок заменил на соответствующие слова

AleksanrReva committed:
Исправляет форму подписки

Исправляет get на post

AleksanrReva committed:
Исправляет разметку заголовков в Контактах

Оставил название магазина в адресе, для индексации и переосмыслил, в принципе без него ничего не ломается. "Время работы" вынес перед параграфом.

AleksanrReva committed:
Исправляет главный заголовок страницы

AleksanrReva committed:
Убирает лишнее у "кнопки" Товара под заказ

AleksanrReva committed:
Добавляет заголовки

Добавил заголовки для тех section, что их не имели

AleksanrReva reviewed (commented):
Внёс необходимые правки

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,358 @@
+
+              <label>
+                <span class="visually-hidden">Найти товар в каталоге</span>
+                <input class="product-search-name" type="text" name="product-name" value="Поиск по сайту">
```
поиск по сайту - это placeholder https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/placeholder

Kartaviy-arteom commented on catalog.html:
```diff
@@ -0,0 +1,358 @@
+          +7 (812) 812-12-12
+        </a>
+
+        <a class="footer-academy-link" href="https://htmlacademy.ru/">
+          .html academy
+        </a>
```
тут будет картинка

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,385 @@
+        <h2 class="visually-hidden">Информация об услугах</h2>
 
-    <p>Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
+        <fieldset class="services-menu">
+          <legend class="visually-hidden">Вид услуги</legend>
```
несколько выглядит непривычно, обчно используют fieldset внутри формы, впринципе можно оставить так

AleksanrReva replied:
и валидатор пропускает. Далее форма не должна понадобиться в этом месте. Если можно оставить, оставляю)

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,385 @@
+      <section class="footer-address">
+        <h2 class="visually-hidden">Адрес нашего интернет-магазина</h2>
+
+        <address>
+          Санкт-Петербург, набережная реки Карповки, 5, литера П.
+        </address>
```
тег address не предназначен для любых почтовых адресов, мы можем его использовать для адреса, если например по этому адресу можем получить услугу. Если это просто какой-то адрес то "p" будет уместен.
Подробнее тут https://html.spec.whatwg.org/multipage/sections.html#the-address-element . Т.е тут ты впринципе можешь его применить, но с ним нужно быть аккуратным

Kartaviy-arteom commented on index.html:
```diff
@@ -4,9 +4,385 @@
+        </a>
+
+        <a class="footer-academy-link" href="https://htmlacademy.ru/">
+          .html academy
```
картинка, это векторный логотип
